tiny_dnn/core/framework/tensor_storage.h

```diff
@@ -68,6 +68,20 @@ static inline size_t product(Container &c) {
                            std::multiplies<size_t>());
 }
 
+template <typename C1, typename C2>
+static inline size_t compute_offset(const C1 &c1, const C2 &c2) {
```

Comment:
The first one could be a range for, and the second a iterator-based one.
Even better, actually: the whole thing could be reduced to a single iteration (adding the constraint for both C1 and C2 to be bidirectional iteratable).